### PR TITLE
Add Syscoin NEVM Chain

### PR DESCRIPTION
**Pull request by osiastedian**
## Description

Add support for [Syscoin Network-Enhanced Virtual Machine](https://docs.syscoin.org/docs/tech/nevm#what-are-virtual-machines).

## Testing instructions

Same as EVM chains

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [ ] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: include/TrustWalletCore/TWCoinType.h [hunk 1]
```diff
@@ -104,6 +104,7 @@ enum TWCoinType {
     TWCoinTypeBoba = 10000288,
     TWCoinTypeMetis = 1001088,
     TWCoinTypeAurora = 1323161554,
+    TWCoinTypeSyscoinNEVM = 57,
```
Comment (catenocrypt) [inline include/TrustWalletCore/TWCoinType.h:107]: For EVM-clones which don't have proper coinID, we derive it from chainID, by adding 10000000, so it should be `10000057`.

File: registry.json [hunk 2]
```diff
@@ -2359,5 +2359,33 @@
       "rpc": "https://mainnet.aurora.dev/",
       "documentation": "https://doc.aurora.dev/"
     }
+  },
+  {
+    "id": "syscoinnevm",
+    "name": "Syscoin NEVM",
+    "coinId": 57,
```
Comment (catenocrypt) [inline registry.json:2366]:
```suggestion
    "coinId": 10000057,
```

File: registry.json [hunk 3]
```diff
@@ -2359,5 +2359,33 @@
...
+    "blockchain": "Ethereum",
+    "derivation": [
+      {
+        "path": "m/44'/60'/0'/0/0"
+      }
+    ],
+    "curve": "secp256k1",
+    "publicKeyType": "secp256k1Extended",
+    "explorer": {
+      "url": "https://explorer.syscoin.org/",
```
Comment (catenocrypt) [inline registry.json:2378]:
```suggestion
      "url": "https://explorer.syscoin.org",
```

**Comment by hewigovens** [comment]
Thanks for your internets, we will consider to add it later

**Pull request closed without merging**